### ISSUE TITLE
crashes with -- 

### ISSUE THREAD

**Issue by hieuhoang** [john-kurkowski/tldextract#191]
I'm getting errors with these 2 urls
    http://www.xn--.ws/
    https://xn--.apk.cafe/
The error is
Traceback (most recent call last):
  File "./temp.py", line 16, in <module>
    ext = tldextract.extract("http://www.xn--.ws/")
  File "/home/hihoan/.local/lib/python3.6/site-packages/tldextract/tldextract.py", line 353, in extract
    return TLD_EXTRACTOR(url)
  File "/home/hihoan/.local/lib/python3.6/site-packages/tldextract/tldextract.py", line 232, in __call__
    translations = [_decode_punycode(label) for label in labels]
  File "/home/hihoan/.local/lib/python3.6/site-packages/tldextract/tldextract.py", line 232, in <listcomp>
    translations = [_decode_punycode(label) for label in labels]
  File "/home/hihoan/.local/lib/python3.6/site-packages/tldextract/tldextract.py", line 405, in _decode_punycode
    return idna.decode(label.encode('ascii')).lower()
  File "/home/hihoan/.local/lib/python3.6/site-packages/idna/core.py", line 391, in decode
    s = ulabel(label)
  File "/home/hihoan/.local/lib/python3.6/site-packages/idna/core.py", line 303, in ulabel
    if label.decode('ascii')[-1] == '-':
IndexError: string index out of range

**Comment by hieuhoang** [comment]
my version is tldextract (2.2.2)

**Comment by commutecat** [comment]
No able to replicate the problem for tldextract 2.2.2.  Did you install from the Pypi store or compile from this git souce? 

```
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tldextract
>>> tldextract.extract("http://www.xn--.ws/")
ExtractResult(subdomain='www', domain='xn--', suffix='ws')

```

**Comment by hieuhoang** [comment]
i installed tldextract using pip3. It seems to occur on Ubuntu 18.04, running on 16.04 is fine 

Ubuntu 18.04 (Python 3.6.9)
>>> tldextract.cli
<module 'tldextract.cli' from '/home/hihoan/.local/lib/python3.6/site-packages/tldextract/cli.py'>

Ubuntu 16.04 (Python 3.5.2)
>>> tldextract.cli
<module 'tldextract.cli' from '/home/hihoan/.local/lib/python3.5/site-packages/tldextract/cli.py'>

**Comment by john-kurkowski** [comment]
Note the uncaught error occurs in 3rd party code, idna. The line isn't in idna 2.8; no error. The line was added in idna 2.9 via kjd/idna#64; yes error.

**Comment by john-kurkowski** [comment]
Opened kjd/idna#82.

**Comment by john-kurkowski** [comment]
Fixed upstream in 2.10.
